### PR TITLE
Update CLI to the newest function CRD

### PR DESCRIPTION
**Pull request by Cortey**
**Description**
Updated CLI to support the newest versions:
- serverless.kyma-project.io/v1alpha2
- gateway.kyma-project.io/v1beta1

**Related issue(s)**
https://github.com/kyma-project/cli/issues/1314

### CODE REVIEW

**Review by clebs — APPROVED** [review]

**Comment by Cortey** [comment]
Waits for https://github.com/kyma-project/hydroform/pull/388

File: go.mod [hunk 1]
```diff
@@ -19,8 +19,8 @@ require (
 	github.com/gardener/component-spec/bindings-go v0.0.66
 	github.com/go-logr/logr v1.2.3
 	github.com/imdario/mergo v0.3.13
-	github.com/kyma-incubator/reconciler v0.0.0-20221108125752-12506041f7e8
-	github.com/kyma-project/hydroform/function v0.0.0-20221021103849-87a95b03cbfa
+	github.com/kyma-incubator/reconciler v0.0.0-20221102095238-a3b6df7d2e27
```
Comment (clebs) [inline go.mod:22]: Please revert this line, the reconciler is being downgraded.

**Review by clebs — CHANGES_REQUESTED** [review]

**Pull request merged**